ddl/ddl_worker.go

```diff
@@ -614,6 +665,144 @@ func (w *JobContext) setDDLLabelForDiagnosis(job *model.Job) {
 	w.ddlJobCtx = kv.WithInternalSourceType(w.ddlJobCtx, w.ddlJobSourceType())
 }
 
+func (w *worker) HandleJobDone(d *ddlCtx, job *model.Job, t *meta.Meta) error {
+	err := w.finishDDLJob(t, job)
+	if err != nil {
+		w.sess.rollback()
+		return err
+	}
+
+	err = w.sess.commit()
+	if err != nil {
+		return err
+	}
+	asyncNotify(d.ddlJobDoneCh)
+	return nil
+}
+
+func (w *worker) HandleDDLJobTable(d *ddlCtx, job *model.Job) error {
+	var (
+		err       error
+		schemaVer int64
+		runJobErr error
+		waitTime  = 2 * d.lease
+	)
+	defer func() {
+		w.unlockSeqNum(err)
+	}()
+
+	err = w.sess.begin()
+	if err != nil {
+		return err
+	}
+	if !variable.EnableConcurrentDDL.Load() {
```

Comment:
Why do we put this check here and not on "Delivery2Worker"?

Reply:
we have to make sure it is true when txn is started